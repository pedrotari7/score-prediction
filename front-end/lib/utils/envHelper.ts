export const backendUrl =
	process.env.NEXT_PUBLIC_APP_ENV === 'local-dev'
		? 'http://localhost:5001/score-prediciton/europe-west1/api/'
		: 'https://europe-west1-score-prediciton.cloudfunctions.net/api';
