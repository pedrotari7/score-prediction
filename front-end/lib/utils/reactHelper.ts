import { DateTime } from 'luxon';

export const classNames = (...classes: any[]) => classes.filter(Boolean).join(' ');

export const getCurrentDate = () => {
	return DateTime.now();
	// Mocked date
	// return DateTime.fromISO('2021-06-16T19:00:00+0000');
};
