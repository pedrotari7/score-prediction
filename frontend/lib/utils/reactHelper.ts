import { DateTime } from 'luxon';
import { Fixture, Prediction, Result, UserResult } from '../../../interfaces/main';

export const classNames = (...classes: string[]) => classes.filter(Boolean).join(' ');

export const getCurrentDate = () => {
	return DateTime.now();
	// Mocked date
	// return DateTime.fromISO('2021-06-16T19:00:00+0000');
	// return DateTime.fromISO('2016-06-09T19:00:00+0000');
};

export const formatScore = (goal: number | null) => {
	if (typeof goal === 'number' && goal >= 0) return goal;
	if (typeof goal === 'number' && goal < 0) return 'H';
	return 'X';
};

export const isNum = (n: number | null) => typeof n === 'number';

export const getOutcome = (g: Result): string | null => {
	if (!isNum(g.home) || !isNum(g.away)) return null;
	if (g.home > g.away) return 'winH';
	if (g.home < g.away) return 'winA';
	if (g.home === g.away) return 'draw';
	return null;
};

export const getResult = (prediction: Prediction, result: Result): Partial<UserResult> => {
	const { home: predH, away: predA } = prediction;
	const { home: realH, away: realA } = result;

	const isExactScore = predH === realH && predA === realA;

	if (isExactScore) return { points: 5, exact: 1 };

	const isCorrectResult =
		!isExactScore && getOutcome(prediction) !== null && getOutcome(prediction) === getOutcome(result);

	if (isCorrectResult) return { points: 3, result: 1 };

	const isCorrectGoal = !isExactScore && !isCorrectResult && (predH === realH || predA === realA);

	if (isCorrectGoal) return { points: 1, onescore: 1 };

	return { points: 0, fail: 1 };
};

export const isGameFinished = (game: Fixture) => ['FT', 'AET', 'PEN'].includes(game.fixture.status.short);
