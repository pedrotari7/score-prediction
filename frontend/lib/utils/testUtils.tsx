import { render } from '@testing-library/react';
import { ReactNode, ContextType } from 'react';
import RouteContext from '../../context/RouteContext';
import NoSpoilersContext from '../../context/NoSpoilersContext';

export const customRender = (
	ui: ReactNode,
	options?: {
		routeContextProps?: ContextType<typeof RouteContext>;
		noSpoilersContextProps?: ContextType<typeof NoSpoilersContext>;
		renderOptions?: Parameters<typeof render>[1];
	}
) => {
	return render(
		<RouteContext.Provider value={options?.routeContextProps ?? null}>
			<NoSpoilersContext.Provider value={options?.noSpoilersContextProps ?? null}>
				{ui}
			</NoSpoilersContext.Provider>
		</RouteContext.Provider>,
		options?.renderOptions
	);
};
