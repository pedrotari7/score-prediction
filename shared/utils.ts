import {
	Competitions,
	Fixture,
	Fixtures,
	Prediction,
	Predictions,
	PredResult,
	Result,
	UserResult,
} from '../interfaces/main';

export const competitions: Competitions = {
	euro2016: { name: 'euro2016', league: 4, season: 2016, start: '2016-06-10', end: '2016-07-10', logo: '/logo.svg' },
	euro2020: { name: 'euro2020', league: 4, season: 2020, start: '2021-06-09', end: '2021-07-15', logo: '/logo.svg' },

	// wc2010: { name: 'wc2010', league: 1, season: 2010, start: '2010-06-10', end: '2010-07-12', logo: '/logo.svg' },
	// wc2014: { name: 'wc2014', league: 1, season: 2014, start: '2014-06-11', end: '2014-07-15', logo: '/logo.svg' },
	// wc2018: { name: 'wc2018', league: 1, season: 2018, start: '2018-06-13', end: '2018-07-16', logo: '/logo.svg' },

	wc2022: {
		name: 'wc2022',
		league: 1,
		season: 2022,
		start: '2022-11-20',
		end: '2022-12-18',
		logo: '/wc2022-logo.svg',
	},
};

export const isNum = (n: number | null): n is number => typeof n === 'number';

export const isEmpty = (obj: Object) => Object.keys(obj).length == 0;

export const getOutcome = (g: Result): 'winH' | 'winA' | 'draw' | null => {
	if (!isNum(g.home) || !isNum(g.away)) return null;
	if (g.home > g.away) return 'winH';
	if (g.home < g.away) return 'winA';
	if (g.home === g.away) return 'draw';
	return null;
};

export const getExtraTimeResult = ({ score: { fulltime, extratime }, fixture, goals }: Fixture) => {
	if (['PEN', 'AET'].includes(fixture.status.short)) {
		return { home: fulltime.home + extratime.home, away: fulltime.away + extratime.away };
	}
	return goals;
};

export const joinResults = (a: Partial<UserResult>, b: Partial<UserResult>) => ({
	...a,
	...b,
	points: (a.points ?? 0) + (b.points ?? 0),
});

export const getResult = (prediction: Prediction, game: Fixture): Partial<UserResult> => {
	const result = getExtraTimeResult(game);

	let ret = {};

	const { home: predH, away: predA } = prediction;
	const { home: realH, away: realA } = result;

	const isExactScore = predH === realH && predA === realA;

	if (isExactScore) ret = joinResults(ret, { points: 5, exact: 1 });

	const isCorrectResult =
		!isExactScore && getOutcome(prediction) !== null && getOutcome(prediction) === getOutcome(result);

	if (isCorrectResult) ret = joinResults(ret, { points: 3, result: 1 });

	const isCorrectGoal = !isExactScore && !isCorrectResult && (predH === realH || predA === realA);

	if (isCorrectGoal) ret = joinResults(ret, { points: 1, onescore: 1 });

	const isPenaltyWinner =
		isPenaltyShootout(game) &&
		getOutcome(prediction) !== null &&
		getOutcome(prediction) === getOutcome(game.score.penalty);

	if (isPenaltyWinner) ret = joinResults(ret, { points: 1, penalty: 1 });

	if (isEmpty(ret)) return { points: 0, fail: 1 };

	return ret;
};

export const isGameFinished = (game: Fixture) => ['FT', 'AET', 'PEN', 'INT', 'PST'].includes(game.fixture.status.short);

export const isPenaltyShootout = (game: Fixture) => game.fixture.status.short === 'PEN';

export const isGameStarted = (game: Fixture) => game.fixture.status.short !== 'NS';

export const isGameOnGoing = (game: Fixture) => !isGameFinished(game) && isGameStarted(game);

export const initializeTeamResult = (): PredResult => ({
	points: 0,
	wins: 0,
	draws: 0,
	loses: 0,
	ga: 0,
	gc: 0,
});

export const calculatePoints = ({ wins, draws }: PredResult) => 3 * wins + draws;

export const calculateResults = (fixtures: Fixture[], predictions: Predictions, uid: string) => {
	const isGroupStage = (f: Fixture) => f.league.round.includes('Group');

	return fixtures.filter(isGroupStage).reduce((teams, game) => {
		const homeTeam = game.teams.home.id;
		const awayTeam = game.teams.away.id;

		if (!(homeTeam in teams)) teams[homeTeam] = initializeTeamResult();
		if (!(awayTeam in teams)) teams[awayTeam] = initializeTeamResult();

		const prediction = predictions?.[game.fixture.id]?.[uid];

		if (prediction?.home >= 0 && prediction?.away >= 0) {
			if (prediction.home > prediction.away) {
				teams[homeTeam].wins += 1;
				teams[awayTeam].loses += 1;
			} else if (prediction?.home < prediction?.away) {
				teams[awayTeam].wins += 1;
				teams[homeTeam].loses += 1;
			} else {
				teams[homeTeam].draws += 1;
				teams[awayTeam].draws += 1;
			}
			teams[homeTeam].points = calculatePoints(teams[homeTeam]);
			teams[awayTeam].points = calculatePoints(teams[awayTeam]);

			teams[homeTeam].ga += prediction?.home ?? 0;
			teams[homeTeam].gc += prediction?.away ?? 0;
			teams[awayTeam].ga += prediction?.away ?? 0;
			teams[awayTeam].gc += prediction?.home ?? 0;
		}

		return teams;
	}, {} as Record<number, PredResult>);
};

export const compareResults = (rA: PredResult, rB: PredResult) => {
	if (rA.points !== rB.points) return rB.points - rA.points;

	const gdA = rA.ga - rA.gc;
	const gdB = rB.ga - rB.gc;

	if (gdA !== gdB) return gdB - gdA;

	if (rA.ga !== rB.ga) return rB.ga - rA.ga;
	return 0;
};

export const sortGroup = (
	groupTeams: number[],
	teamsResults: Record<number, PredResult>,
	fixtures: Fixtures,
	predictions: Predictions,
	userID: string
) => {
	const teamsWithSamePoints = groupTeams.reduce<Record<number, number[]>>((pack, teamID) => {
		const points = teamsResults[teamID].points;
		if (!(points in pack)) pack[points] = [];
		pack[points].push(teamID);
		return pack;
	}, {});

	const sortedStandings = Object.entries(teamsWithSamePoints)
		.sort(([a], [b]) => parseInt(b) - parseInt(a))
		.flatMap(([_, teams]) => {
			if (teams.length === 1) return teams;

			const tiedTeamsGames = Object.values(fixtures).filter(
				(f: Fixture) => teams.includes(f.teams.home.id) && teams.includes(f.teams.away.id)
			);

			const tiedTeamsResults = calculateResults(tiedTeamsGames, predictions, userID);

			const sortedTeams = Object.entries(tiedTeamsResults)
				.sort(([teamA, resultsA], [teamB, resultsB]) => {
					const localResult = compareResults(resultsA, resultsB);

					if (localResult) return localResult;

					return compareResults(teamsResults[parseInt(teamA)], teamsResults[parseInt(teamB)]);
				})
				.map(([t]) => parseInt(t));

			return sortedTeams;
		});

	return sortedStandings;
};
