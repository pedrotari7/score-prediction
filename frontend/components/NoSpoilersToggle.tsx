import { EyeIcon, EyeSlashIcon } from '@heroicons/react/24/outline';
import Switch from 'react-switch';
import React from 'react';
import 'react-toggle/style.css'; // for ES6 modules
import useNoSpoilers from '../hooks/useNoSpoilers';
import { classNames } from '../lib/utils/reactHelper';

const NoSpoilersToggle = () => {
	const { noSpoilers, setNoSpoilers } = useNoSpoilers();

	return (
		<label
			className={classNames(
				'flex flex-col items-center justify-between gap-2 rounded-md py-2 text-xs font-bold sm:py-4',
				'cursor-pointer'
			)}
		>
			<Switch
				onChange={setNoSpoilers}
				onColor='#181a1b'
				uncheckedIcon={
					<div className='flex size-full items-center justify-center'>
						<EyeIcon strokeWidth={2} className='size-4 text-white' />
					</div>
				}
				checkedIcon={
					<div className='flex size-full items-center justify-center'>
						<EyeSlashIcon strokeWidth={2} className='size-4 text-white' />
					</div>
				}
				checked={noSpoilers ?? false}
				// eslint-disable-next-line tailwindcss/no-custom-classname
				className='react-switch'
			/>
			<span>No Spoilers</span>
		</label>
	);
};

export default NoSpoilersToggle;
