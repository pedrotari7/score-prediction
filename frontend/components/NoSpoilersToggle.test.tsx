import { act } from '@testing-library/react';
import { customRender } from '../lib/utils/testUtils';
import NoSpoilersToggle from './NoSpoilersToggle';
import type { ContextType } from 'react';
import type NoSpoilersContext from '../context/NoSpoilersContext';

describe('NoSpoilersToggle', () => {
	it('renders no spoilers toggle turned off if no context', () => {
		const { container } = customRender(<NoSpoilersToggle />, {
			noSpoilersContextProps: { noSpoilers: undefined, setNoSpoilers: () => {} } as unknown as ContextType<
				typeof NoSpoilersContext
			>,
		});
		expect(container).toMatchSnapshot();
	});

	it('renders no spoilers toggle turned off', () => {
		const { container } = customRender(<NoSpoilersToggle />, {
			noSpoilersContextProps: { noSpoilers: false, setNoSpoilers: () => {} },
		});
		expect(container).toMatchSnapshot();
	});

	it('renders no spoilers toggle turned on', () => {
		const { container } = customRender(<NoSpoilersToggle />, {
			noSpoilersContextProps: { noSpoilers: true, setNoSpoilers: () => {} },
		});
		expect(container).toMatchSnapshot();
	});

	it('toggle switch and update context', () => {
		const noSpoilersContextProps = { noSpoilers: false, setNoSpoilers: jest.fn() };
		const { container } = customRender(<NoSpoilersToggle />, { noSpoilersContextProps });

		expect(container).toMatchSnapshot();

		const toggle = container.querySelector('input');

		act(() => {
			if (toggle) {
				toggle.click();
			}
		});

		expect(noSpoilersContextProps.setNoSpoilers).toHaveBeenCalledTimes(1);
		expect(noSpoilersContextProps.setNoSpoilers).toHaveBeenCalledWith(true, expect.anything(), undefined);
	});
});
