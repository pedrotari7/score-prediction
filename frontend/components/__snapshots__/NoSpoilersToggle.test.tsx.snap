// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSpoilersToggle renders no spoilers toggle turned off 1`] = `
<div>
  <label
    class="flex flex-col items-center justify-between gap-2 rounded-md py-2 text-xs font-bold sm:py-4 cursor-pointer"
  >
    <div
      class="react-switch"
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(136, 136, 136); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
      >
        <div
          style="height: 28px; width: 30px; position: relative; opacity: 0; pointer-events: none; transition: opacity 0.25s;"
        >
          <div
            class="flex  size-full items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="size-4 text-white"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.98 8.223A10.477 10.477 0 0 0 1.934 12C3.226 16.338 7.244 19.5 12 19.5c.993 0 1.953-.138 2.863-.395M6.228 6.228A10.451 10.451 0 0 1 12 4.5c4.756 0 8.773 3.162 10.065 7.498a10.522 10.522 0 0 1-4.293 5.774M6.228 6.228 3 3m3.228 3.228 3.65 3.65m7.894 7.894L21 21m-3.228-3.228-3.65-3.65m0 0a3 3 0 1 0-4.243-4.243m4.242 4.242L9.88 9.88"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
        <div
          style="height: 28px; width: 30px; position: absolute; opacity: 1; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
        >
          <div
            class="flex size-full items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="size-4 text-white"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.036 12.322a1.012 1.012 0 0 1 0-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="react-switch-handle"
        style="height: 26px; width: 26px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        aria-checked="false"
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`NoSpoilersToggle renders no spoilers toggle turned off if no context 1`] = `
<div>
  <label
    class="flex flex-col items-center justify-between gap-2 rounded-md py-2 text-xs font-bold sm:py-4 cursor-pointer"
  >
    <div
      class="react-switch"
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(136, 136, 136); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
      >
        <div
          style="height: 28px; width: 30px; position: relative; opacity: 0; pointer-events: none; transition: opacity 0.25s;"
        >
          <div
            class="flex  size-full items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="size-4 text-white"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.98 8.223A10.477 10.477 0 0 0 1.934 12C3.226 16.338 7.244 19.5 12 19.5c.993 0 1.953-.138 2.863-.395M6.228 6.228A10.451 10.451 0 0 1 12 4.5c4.756 0 8.773 3.162 10.065 7.498a10.522 10.522 0 0 1-4.293 5.774M6.228 6.228 3 3m3.228 3.228 3.65 3.65m7.894 7.894L21 21m-3.228-3.228-3.65-3.65m0 0a3 3 0 1 0-4.243-4.243m4.242 4.242L9.88 9.88"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
        <div
          style="height: 28px; width: 30px; position: absolute; opacity: 1; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
        >
          <div
            class="flex size-full items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="size-4 text-white"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.036 12.322a1.012 1.012 0 0 1 0-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="react-switch-handle"
        style="height: 26px; width: 26px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        aria-checked="false"
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`NoSpoilersToggle renders no spoilers toggle turned on 1`] = `
<div>
  <label
    class="flex flex-col items-center justify-between gap-2 rounded-md py-2 text-xs font-bold sm:py-4 cursor-pointer"
  >
    <div
      class="react-switch"
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(24, 26, 27); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
      >
        <div
          style="height: 28px; width: 30px; position: relative; opacity: 1; pointer-events: none; transition: opacity 0.25s;"
        >
          <div
            class="flex  size-full items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="size-4 text-white"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.98 8.223A10.477 10.477 0 0 0 1.934 12C3.226 16.338 7.244 19.5 12 19.5c.993 0 1.953-.138 2.863-.395M6.228 6.228A10.451 10.451 0 0 1 12 4.5c4.756 0 8.773 3.162 10.065 7.498a10.522 10.522 0 0 1-4.293 5.774M6.228 6.228 3 3m3.228 3.228 3.65 3.65m7.894 7.894L21 21m-3.228-3.228-3.65-3.65m0 0a3 3 0 1 0-4.243-4.243m4.242 4.242L9.88 9.88"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
        <div
          style="height: 28px; width: 30px; position: absolute; opacity: 0; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
        >
          <div
            class="flex size-full items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="size-4 text-white"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.036 12.322a1.012 1.012 0 0 1 0-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="react-switch-handle"
        style="height: 26px; width: 26px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(29px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        aria-checked="true"
        checked=""
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`NoSpoilersToggle toggle switch and update context 1`] = `
<div>
  <label
    class="flex flex-col items-center justify-between gap-2 rounded-md py-2 text-xs font-bold sm:py-4 cursor-pointer"
  >
    <div
      class="react-switch"
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(136, 136, 136); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
      >
        <div
          style="height: 28px; width: 30px; position: relative; opacity: 0; pointer-events: none; transition: opacity 0.25s;"
        >
          <div
            class="flex  size-full items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="size-4 text-white"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.98 8.223A10.477 10.477 0 0 0 1.934 12C3.226 16.338 7.244 19.5 12 19.5c.993 0 1.953-.138 2.863-.395M6.228 6.228A10.451 10.451 0 0 1 12 4.5c4.756 0 8.773 3.162 10.065 7.498a10.522 10.522 0 0 1-4.293 5.774M6.228 6.228 3 3m3.228 3.228 3.65 3.65m7.894 7.894L21 21m-3.228-3.228-3.65-3.65m0 0a3 3 0 1 0-4.243-4.243m4.242 4.242L9.88 9.88"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
        <div
          style="height: 28px; width: 30px; position: absolute; opacity: 1; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
        >
          <div
            class="flex size-full items-center justify-center"
          >
            <svg
              aria-hidden="true"
              class="size-4 text-white"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.036 12.322a1.012 1.012 0 0 1 0-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="react-switch-handle"
        style="height: 26px; width: 26px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        aria-checked="false"
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
  </label>
</div>
`;
